platform/mds/mds/src/main/java/org/motechproject/mds/annotations/Discriminated.java

```diff
@@ -0,0 +1,13 @@
+package org.motechproject.mds.annotations;
+
+import javax.jdo.annotations.IdentityType;
+import javax.jdo.annotations.PersistenceCapable;
+import java.lang.annotation.*;
+
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@PersistenceCapable(identityType = IdentityType.DATASTORE, detachable = "true")
+public @interface Discriminated {
```

Comment:
I'm wondering this because we do have some general support for the regular JDO inheritance. In general we would use the regular JDO annotations unless we wanted our spin on it, for example with the entity annotation you get the databrowser, history etc. But on the other hand we just use the regular JDO unique annotation for annotating a column you want a unique constraint on. The main reasoning was that going with all new annotations would just be reinventing the wheel and needless work. I will admit though that we should consider moving some stuff to our annotations for clarity, "defaultFetchGroup" in @PersistenceCapable is an example. But back on point, I'm just wondering how we go forth with inheritance with MDS, will this be the recommended way, the only way or just an optional way to do inheritance?

Regarding the benefits, if you would ask my why use Entity instead of PersistenceCapable, I would say that you get the databrowser, history tracking, REST API and all that good stuff. Now, if I would ask use why use the new annotation instead of regular JDO inheritance, what would be the anwser? I primarily see UI benefits, namely the enhanced data browser support - a solution to that message campaing issue that was hotfixed for Ebodac using javascript customization. Is that correct or am I missing something?

As I mentioned, I see the value of this, I am just wondering how we go forward with inheritance support. Do we keep the two ways separate or do we merge them somehow?